src/stores/manager/service.go

```diff
@@ -125,64 +121,58 @@ func (m *BaseManager) loadAll() {
+func (m *BaseManager) GetSecretStore(ctx context.Context, storeName string, userInfo *authtypes.UserInfo) (stores.SecretStore, error) {
 	m.mux.RLock()
 	defer m.mux.RUnlock()
 
-	if storeBundle, ok := m.secrets[name]; ok {
-		if store, ok := storeBundle.store.(secrets.Store); ok {
+	if storeBundle, ok := m.secrets[storeName]; ok {
+		if store, ok := storeBundle.store.(stores.SecretStore); ok {
 			policies := m.policyManager.UserPolicies(ctx, userInfo)
-			resolvr, err := policy.NewResolver(policies)
-			if err != nil {
-				return nil, err
-			}
-			return secretsconnector.NewSecretConnector(store, resolvr, storeBundle.logger), nil
+			_, _ = policy.NewResolver(policies)
+			return secretsconnector.NewSecretConnector(store, m.db.Secrets(storeName), storeBundle.logger), nil
 		}
 	}
 
 	errMessage := "secret store was not found"
-	m.logger.Error(errMessage, "store_name", name)
+	m.logger.Error(errMessage, "store_name", storeName)
 	return nil, errors.NotFoundError(errMessage)
 }
 
-func (m *BaseManager) GetKeyStore(ctx context.Context, name string, userInfo *authtypes.UserInfo) (connectors.KeysConnector, error) {
+func (m *BaseManager) GetKeyStore(_ context.Context, storeName string, _ *authtypes.UserInfo) (stores.KeyStore, error) {
 	m.mux.RLock()
 	defer m.mux.RUnlock()
-	if storeBundle, ok := m.keys[name]; ok {
-		if store, ok := storeBundle.store.(keys.Store); ok {
-			return keysconnector.NewConnector(store, m.db, storeBundle.logger), nil
+	if storeBundle, ok := m.keys[storeName]; ok {
+		if store, ok := storeBundle.store.(stores.KeyStore); ok {
+			return keysconnector.NewConnector(store, m.db.Keys(storeName), storeBundle.logger), nil
 		}
 	}
 
 	errMessage := "key store was not found"
-	m.logger.Error(errMessage, "store_name", name)
+	m.logger.Error(errMessage, "store_name", storeName)
 	return nil, errors.NotFoundError(errMessage)
 }
 
-func (m *BaseManager) GetEth1Store(ctx context.Context, name string, userInfo *authtypes.UserInfo) (eth1store.Store, error) {
+func (m *BaseManager) GetEth1Store(ctx context.Context, name string, userInfo *authtypes.UserInfo) (stores.Eth1Store, error) {
 	m.mux.RLock()
 	defer m.mux.RUnlock()
 	return m.getEth1Store(ctx, name, userInfo)
 }
 
-func (m *BaseManager) getEth1Store(ctx context.Context, name string, userInfo *authtypes.UserInfo) (eth1store.Store, error) {
-	if storeBundle, ok := m.eth1Accounts[name]; ok {
-		if store, ok := storeBundle.store.(eth1store.Store); ok {
+func (m *BaseManager) getEth1Store(ctx context.Context, storeName string, userInfo *authtypes.UserInfo) (stores.Eth1Store, error) {
+	if storeBundle, ok := m.eth1Accounts[storeName]; ok {
+		if store, ok := storeBundle.store.(stores.KeyStore); ok {
 			policies := m.policyManager.UserPolicies(ctx, userInfo)
-			resolvr, err := policy.NewResolver(policies)
-			if err != nil {
-				return nil, err
-			}
-			return eth1.NewEth1Connector(store, resolvr, storeBundle.logger), nil
+			_, _ = policy.NewResolver(policies)
```

Comment:
why we don't care about what NewResolver does?

Reply:
PolicyResolver will be implemented as part of this ticket #237